File: Java/pattern/builder/src/com/pattern/Director.java
package com.pattern;

public class Director {
	public Director(IBuilder myBuilder) {
		super();
		this.myBuilder = myBuilder;
	}
	
	public void create() {
		myBuilder.createPart1();
		myBuilder.createPart2();
	}

	private IBuilder myBuilder;
	
}

File: cplusplus/design_pattern/builderProj/Director.cpp
#include "stdafx.h"
#include "Director.h"
#include "IBuilder.h"


Director::Director()
{
    //std::cout << "Director::Director()" << std::endl;
}


Director::~Director()
{
    //std::cout << "Director::~Director()" << std::endl;
}


Director::Director(std::shared_ptr<IBuilder> builder) : m_Builder(builder)
{
    //std::cout << "Director::Director(std::shared_ptr<IBuilder> builder)" << std::endl;
}


void Director::create()
{
    m_Builder->createPart1();
    m_Builder->createPart2();
}
